### DISCUSSION TITLE
Can't use PHPMailer to sign with S/MIME, getting "Signing Error: "

### DISCUSSION THREAD

**Discussion by flaggalagga** [PHPMailer/PHPMailer#2674] [Q&A] [answered]
I'm using phpmailer to try to send signed emails (S/MIME).

I used this excellent tutorial by John Dalesandro [https://www.dalesandro.net/create-self-signed-smime-certificates/](url) to create a Self-Signed S/MIME Certificate and package it in PKCS12 Format.

I then followed the instructions in PHPMailer S/MIME signing [https://github.com/PHPMailer/PHPMailer/blob/master/examples/smime_signed_mail.phps](url) to make **cert.crt, cert.key & certchain.pem** from my PKCS12 package.

When I sign with the code below I get "Signing Error:" :

```
$mail->sign(
  '/mypath/cert.crt', 
  '/mypath/cert.key', 
  'mypassword',
  '/mypath/certchain.pem'
);
```

The emails are sent normally when I don't try to sign the email.

Since there is no explanation at all in my error code, I don't understand what I'm doing wrong.

**Comment by flaggalagga** [comment, marked as answer]
So I got a S/MIME certificate in PKCS12 from a external certificate authority. I then made a **cert.crt, cert.key & certchain.pem** as stated above and now it works.

**Reply by Synchro** [reply]
Thanks for following up on this, and I'm glad you worked it out!
